pymoab/range.pyx

```diff
@@ -35,4 +35,7 @@ cdef class Range(object):
 
     def __getitem__(self, int index):
         cdef moab.EntityHandle rtn = deref(self.inst)[index]
-        return rtn
+        if index < self.size():
+            return rtn
+        else:
+            raise StopIteration
```

Comment:
Hmm this is a weird place for the StopIteration.  Typically this happens in the `__iter__()` method, which seems to be missing.  I would suggest this method going back to its original state and then adding the following `__iter__()`

```python
def __iter__(self):
    cdef int i = 0
    for i in range(0, self.inst.size()):
        yield self[i]
```